### ISSUE TITLE
Current `store` interface is confusing when using `Bytes` `id`

### ISSUE THREAD

**Issue by lutter** [graphprotocol/graph-node#3493]
The functions in the `store` module (`get`, `set`, `remove`) are a bit confusing right now when subgraph use `Bytes` as their `id`; the host interface still requires that the `id` is passed as a string, for `Bytes` it needs to be `id.toHexString()`. For `get` and `set`, the difference is hidden by the code that `graph-cli` generates, so that subgraph authors can say `Thing.get(id)` no matter whether `Thing` uses `String` or `Bytes` for its ids.

But for `remove`, users need to use the 'raw' interface and write `store.remove(id.toHexString())` which can be confusing.

I don't think we should actually change the calling convention for the `store` functions; instead, we should document this more clearly and explain how to work with these functions when `id` has type `Bytes`.

This was first reported as part of #3476

**Comment by azf20** [comment]
Nice - would this primarily be an update [here](https://thegraph.com/docs/en/developer/assemblyscript-api/#removing-entities-from-the-store), or is this also about documentation within the graph-node repo?

**Comment by lutter** [comment]
> Nice - would this primarily be an update [here](https://thegraph.com/docs/en/developer/assemblyscript-api/#removing-entities-from-the-store), or is this also about documentation within the graph-node repo?

No, I think that would be all in the docs repo; not really in `graph-node` - so we should probably move the issue?
